rufio/internal/controller/client.go

```diff
@@ -232,3 +252,22 @@ func toStringSlice(p []bmc.ProviderName) []string {
 	}
 	return s
 }
+
+// createHTTPClientWithProxy creates an HTTP client configured to use the specified proxy.
+func createHTTPClientWithProxy(proxyURL string, insecureTLS bool) *http.Client {
```

Comment:
by creating our own `http.Client` we lose settings/configuration that bmclib does for its default http client. [Reference](https://github.com/bmc-toolbox/bmclib/blob/main/internal/httpclient/httpclient.go). We might want to copy this over here.